### PR TITLE
Eherot/support deleting servers

### PR DESCRIPTION
**Pull request by eherot**

### CODE REVIEW

File: Berksfile [hunk 1]
```diff
@@ -1,13 +1,12 @@
-source 'https://api.berkshelf.com'
+source 'https://supermarket.chef.io'
+source 'https://berks.evertrue.com'
```
Comment (jeffbyrnes) [inline Berksfile]: Curious; isn't this supposed to be an OSS cookbook?

Comment (eherot) [inline Berksfile]: Technically yes but the testing requires the hostnames to be set up and `et_hostname` is not currently open sourced (contains a bunch of proprietary BS).

File: recipes/definitions.rb [hunk 2]
```diff
@@ -88,10 +88,27 @@
...
 end
 
-search(
+active_hosts_list = search(
   :node,
   node['shinken']['host_search_query']
-).each do |n|
+)
+
+if Dir.exist?('/etc/shinken/hosts')
```
Comment (jeffbyrnes) [inline recipes/definitions.rb]: Lotta uses of this path; might be good to turn it into a local var, at least, or potentially an attribute?

File: recipes/definitions.rb [hunk 3]
```diff
@@ -88,10 +88,27 @@
...
+  current_hosts_list = (Dir.entries('/etc/shinken/hosts') - ['.', '..']).map do |e|
```
Comment (jeffbyrnes) [inline recipes/definitions.rb]: Might be cleaner to use `Dir.glob` instead?

```ruby
Dir.glob("#{dirname}/*").each do |fname|
  puts fname
end
```

**Comment by jeffbyrnes** [comment]
@eherot solid stuff! A few Ruby suggestions, but the logic appears sound.

Curious; is there any blipping or other issue with restarting the `shinken-arbiter` like this?

**Comment by eherot** [comment]
Shouldn't be. The arbiter doesn't handle the pinging or the UI.

File: recipes/definitions.rb (hunk 2 above)
Comment (eherot) [inline recipes/definitions.rb]: I'll do this directly on master.

File: recipes/definitions.rb (hunk 3 above)
Comment (eherot) [inline recipes/definitions.rb]: Dir.glob gets whole path names, so there would be a lot of other cleaning to do.

**Pull request merged**